dashboard/test/controllers/api/v1/pd/workshop_attendance_controller_test.rb

```diff
@@ -3,214 +3,302 @@
-      assert_response :success
+  test 'create attendance allows admins to override and add teachers to the section' do
+    sign_in create(:admin)
+    teacher = create :teacher
+
+    # set_attendance is idempotent
+    assert_creates Pd::Attendance, Follower do
+      2.times do
+        create_attendance @workshop, @session, teacher, admin_override: 1
+        assert_response :success
+      end
     end
-    assert_equal 1, Pd::Attendance.for_teacher(User.last).for_workshop(@workshop).count
-    assert @workshop.section.students.exists?(User.last.id)
+    assert_equal teacher.id, Pd::Attendance.last.teacher_id
+    assert_equal @session.id, Pd::Attendance.last.pd_session_id
+    assert_equal teacher.id, Follower.last.student_user_id
   end
 
-  test 'admin creating a user without enrollment raises error' do
+  test 'create attendance returns 404 even with admin_override when the teacher does not exist' do
     sign_in create(:admin)
-    email = "new_teacher#{SecureRandom.hex(4)}@example.net"
 
-    update_attendance @workshop.id, create_user_params(email)
+    create_attendance @workshop, @session, build(:user, id: -1), admin_override: 1
     assert_response :not_found
   end
 
-  test 'non-admins cannot create new users' do
+  test 'delete_attendance' do
     sign_in @organizer
-    email = "new_teacher#{SecureRandom.hex(4)}@example.net"
-    create :pd_enrollment, workshop: @workshop, email: email
+    teacher = create :teacher
+    @workshop.section.add_student teacher
+    create :pd_attendance, session: @session, teacher: teacher
 
-    update_attendance @workshop.id, create_user_params(email)
-    assert_response :forbidden
+    # delete_attendance is idempotent
+    2.times do
+      delete_attendance @workshop, @session, teacher
+      assert_response :success
+    end
+
+    refute Pd::Attendance.where(session: @session, teacher: teacher).exists?
   end
 
-  test 'section_code' do
+  test 'create delete and create restores the original record' do
     sign_in @organizer
-    get_attendance @workshop.id
-    response = JSON.parse(@response.body)
-    assert_nil response['section_code']
+    teacher = create :teacher
+    @workshop.section.add_student teacher
 
-    # Start the workshop, creating a section
-    @workshop.start!
-    get_attendance @workshop.id
-    response = JSON.parse(@response.body)
-    assert_equal @workshop.reload.section.code, response['section_code']
+    create_attendance @workshop, @session, teacher
+    assert_response :success
+    attendance = Pd::Attendance.last
+
+    delete_attendance @workshop, @session, teacher
+    assert_response :success
+    assert attendance.reload.persisted?
+    assert attendance.deleted?
+
+    create_attendance @workshop, @session, teacher
+    assert_response :success
+    refute attendance.reload.deleted?
   end
 
   private
 
+  def assert_manage_response(response, workshop: self.workshop, session: nil, user: self.teacher)
```

Comment:
weird. All I did was rebase staging. I wonder why Hound decided to complain now, when this line didn't change. Will fix...